components/formats-api/src/loci/formats/FormatWriter.java

```diff
@@ -293,6 +293,42 @@ public void setWriteSequentially(boolean sequential) {
     this.sequential = sequential;
   }
 
+  /* @see IFormatWriter#getTileSizeX() */
+  @Override
+  public int getTileSizeX() throws FormatException {
+    PositiveInteger width = metadataRetrieve.getPixelsSizeX(getSeries());
+    if (width == null) throw new FormatException("Pixels Size X must not be null when attempting to get tile size.");
+    return width.getValue();
+  }
+
+  /* @see IFormatWriter#setTileSizeX(int) */
+  @Override
+  public int setTileSizeX(int tileSize) throws FormatException {
+    PositiveInteger width = metadataRetrieve.getPixelsSizeX(getSeries());
+    if (width == null) throw new FormatException("Pixels Size X must not be null when attempting to set tile size.");
+    if (tileSize <= 0) throw new FormatException("Tile size must be > 0.");
+    if (tileSize > width.getValue()) throw new FormatException("Tile width must be <= image width (" + width.getValue() + ")");
```

Comment:
Why? Surely we already have to deal with the last column of tiles being wider than what remains of an image?

Reply:
I'm indeed now running into this. Naively trying to set tile size in advance in the hope of not needing my tile-size-setting per-plane IFDs any more fails because I apparently also now need to set this pixels size thing somehow.